IliasDownloaderUniMA/__init__.py

```diff
@@ -100,6 +120,29 @@ def login(self, login_id, login_pw):
 			raise ConnectionError("Couldn't log into ILIAS. Make sure your provided uni-id and the password are correct.")
 
 
+	def extractUserCourses(self):
+		"""
+		Extracts the users subscribed courses from the login_soup
+
+		:returns:   List of dicts, with keys name, url and iliasid
+		:rtype:     list
+		"""
+
+		courses = []
+
+		containers = self.login_soup.select("div.il_ContainerListItem")
+		for container in containers:
+			title_elem = container.select("a.il_ContainerItemTitle")[0]
+
+			course_name = self.cleanCourseName(title_elem.string)
+			iliasid = self.extractIdFromUrl(title_elem["href"])
+
+			url = self.createIliasUrl(iliasid)
+			courses += [{"name": course_name, "url": url, "iliasid": iliasid}]
+
+		return courses
```

Comment:
```suggestion
	def addAllCurrentSemesterCourses(self):
		"""
		Adds all current semester courses.
		"""

		items = self.login_soup.find_all("a", "il_ContainerItemTitle")

		# Find the current semester (should be the first entry
		# matching the regex pattern)
		for i in items:
			match = re.search(r"([A-Z]{3} \d{4})", i.text)
			if match:
				current_semester = match.group(0)
				break
		# add the current semester's courses
		for course in items:
			if current_semester in course.text:
				url = course['href']
				ilias_id = int(re.search(r"ref_id=(\d+)", url).group(1))
				self.addCourse(ilias_id)
```

Reply:
See my comments to your proposals for extractUserCourses and addAllCourses above, plus my design idea below.
I don't like this change tbh. It removes the excluded_ids, works only for the current semester and will make things a lot harder to test.
I'd prefer a flexible over a rigid design.